### ISSUE TITLE
[BUG] `SeasonalityACF` fails in presence of multiple time series - multivariate targets or panel data

### ISSUE THREAD

**Issue by yarnabrina** [sktime/sktime#4845]
Reproducible Code

```python
from sktime.param_est.seasonality import SeasonalityACF
from sktime.transformations.series.difference import Differencer
pipeline = Differencer() * SeasonalityACF()

from sktime.datasets import load_airline
y = load_airline()

pipeline.fit(y)  # works
# ValueError: input must be univariate pd.DataFrame, with one column

from sktime.datasets import load_longley
_, X = load_longley()

pipeline.fit(X)  # fails
# ValueError: input must be univariate pd.DataFrame, with one column

from sktime.utils._testing.hierarchical import _make_hierarchical
df = _make_hierarchical()

pipeline.fit(df)  # fails
# TypeError: X must be in an sktime compatible format, of scitypes ['Series'], for instance a pandas.DataFrame with sktime compatible time indices, or with MultiIndex and last(-1) level an sktime compatible time index. See data format tutorial examples/AA_datatypes_and_datasets.ipynb,If you think X is already in an sktime supported input format, run sktime.datatypes.check_raise(X, mtype) to diagnose the error, where mtype is the string of the type specification you want for y. Possible mtype specification strings are as follows. "For Series scitype: ['pd.DataFrame', 'pd.Series', 'np.ndarray'].
```

Version: `0.20.0` on `macOS Ventura`

**Comment by fkiraly** [comment]
Hm, I wouldn't exactly classify this as a bug but a missing feature.

For context:

* forecasters, transformers broadcast/vectorize over multiple instances and variables
* parameter estimators currently do not

Now, the pipeline `pipe` is of type "parameter estimator", so will fail to vectorize.

However, if you would pipe if the other way round, e.g., with a `PluginParamsForecaster`, you would obtain a forecaster that would broadcast.

**Comment by fkiraly** [comment]
possible actionables:

* plugin of parameter estimator into transformer
* dunder for plugin (param_est * trafo or param_est * forecaster)
* broadcasting for parameter estimators

**Comment by yarnabrina** [comment]
Agreed on all three. I suppose we can consider this issue as an umbrella issue for enhance tasks for parameter estimators.

Quick question, if `param_est * transformer` is supported, will it be enough to bahve as if it supports broadcasting in all forms of pipelines?

I'm asking because I'm not sure what would be the expected output format if just param_est is applied directly on a dataset, without any pipeline. A dataframe of same number of columns as input and same number of rows as lowest level of hierarchy?

Can there be a case that user would want estimates based on higher level, which may have a different behavious upon aggregation?

**Comment by fkiraly** [comment]
> will it be enough to bahve as if it supports broadcasting in all forms of pipelines?

Yes - the first step would be implementing a `PluginParamsTransformer`, I think.

> I'm asking because I'm not sure what would be the expected output format if just param_est is applied directly on a dataset, without any pipeline. A dataframe of same number of columns as input and same number of rows as lowest level of hierarchy?

I don't know, `get_fitted_params` would have to do sth sensible.

Something like current forecaster broadcasting in `get_fitted_params`, or something like this? https://github.com/sktime/sktime/pull/4107

> Can there be a case that user would want estimates based on higher level, which may have a different behavious upon aggregation?

Not sure, can you explain?

**Comment by yarnabrina** [comment]
I don't know why I missed this. What I meant is that (store-item) level data may present one type of seasonality than (store) level data when aggregated across all items.

---

I somewhat need this broadcasting feature. I posted some question in discord earlier in the morning regarding how to debug inside a pipeline, as I faced something is working and just by changing dataset it started failing. ANyway, I think this is the related feature.

I want to contribute to add at least some of these functionalities, but I'm not familiar with dunder methods or broadcasting logics much. Can you please suggest where should I start looking for exising functionalities, and where do I need make changes for these actionable? Also, if you can guide about easiest to start with etc. that will be very helpful for me to get familiar.

> * plugin of parameter estimator into transformer
> * dunder for plugin (param_est * trafo or param_est * forecaster)
> * broadcasting for parameter estimators

**Comment by fkiraly** [comment]
Ah, sorry, I also forgot about this!

Re how to get started, I think the plugin of estimator into transformer would solve this?

The actionable would be:

1. look at `PluginParamsForecaster`
2. make a `PluginParamsTransformer` along the same lines
3. possibly refactor the module (e.g., into submodules, or factor out the parameter logic in the `_fit`-s)

The dunders would be syntactic sugar, the actionables here would be:

1. look at the `__mul__` and `__rmul__` dunders of transformers
2. I would give the `PluginParamsForecaster` a `__mul__` dunder along thes lines, to make right multiplication a plugin.
(although not sure how we could incorporate renames)

**Comment by yarnabrina** [comment]
@fkiraly is this closed by #5764 and will be released in 0.25.1 or 0.26?

**Comment by fkiraly** [comment]
hmmm, possibly - would you like to try this on your use case and report whether it is fine for you?

**Comment by yarnabrina** [comment]
I just ran the issue code on `main` branch, and it failed for me with
identical errors as reported above.

Environment:

Docker: python:3.10-slim-bullseye
Install: pip install ***@***.***, pip
install statsmodels

**Comment by fkiraly** [comment]
it should work though if you then plug it into, say, `Deseasonalizer` - i.e., if the final result is a transformer.

That would deseasonalize each variable by its most likely seasonality.

**Comment by fkiraly** [comment]
This works now:

```python
from sktime.param_est.seasonality import SeasonalityACF
from sktime.transformations.series.detrend import Deseasonalizer
from sktime.transformations.series.difference import Differencer
from sktime.utils._testing.hierarchical import _make_hierarchical

X = _make_hierarchical()

sp_auto = Differencer() * SeasonalityACF() * Deseasonalizer()

sp_auto.fit(X)
```
